internal/adapters/entrypoints/rest/handlers/get_reports_revenue_test.go

```diff
@@ -0,0 +1,270 @@
+			result: http.StatusInternalServerError,
+		},
+		{
+			name:      "should return 200 if use case succeeds",
+			startDate: "2024-01-01",
+			endDate:   "2025-08-27",
+			mockSetup: func(useCase *MockGetRevenueReportUseCase) {
+				useCase.On("Run", mock.Anything, mock.Anything, mock.Anything).
+					Return(reports.GetRevenueReportResult{
+						TotalQuoteCallFees:    entities.NewWei(1000),
+						TotalGasFeesCollected: entities.NewWei(500),
+						TotalGasSpent:         entities.NewWei(300),
+						TotalPenalizations:    entities.NewWei(100),
+						TotalProfit:           entities.NewWei(1100),
+					}, nil).Once()
+			},
+			result: http.StatusOK,
+		},
+		{
+			name:      "should return 200 with zero values when no quotes exist",
+			startDate: "2024-01-01",
+			endDate:   "2024-01-02",
+			mockSetup: func(useCase *MockGetRevenueReportUseCase) {
+				useCase.On("Run", mock.Anything, mock.Anything, mock.Anything).
+					Return(reports.GetRevenueReportResult{
+						TotalQuoteCallFees:    entities.NewWei(0),
+						TotalGasFeesCollected: entities.NewWei(0),
+						TotalGasSpent:         entities.NewWei(0),
+						TotalPenalizations:    entities.NewWei(0),
+						TotalProfit:           entities.NewWei(0),
+					}, nil).Once()
+			},
+			result: http.StatusOK,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			useCase := &MockGetRevenueReportUseCase{}
+			tc.mockSetup(useCase)
+
+			req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "/reports/revenue", nil)
+			require.NoError(t, err)
+
+			q := req.URL.Query()
+			if tc.startDate != "" {
+				q.Add("startDate", tc.startDate)
+			}
+			if tc.endDate != "" {
+				q.Add("endDate", tc.endDate)
+			}
+			req.URL.RawQuery = q.Encode()
+			rr := httptest.NewRecorder()
+
+			handler := handlers.NewGetReportsRevenueHandler(useCase)
+
+			handler.ServeHTTP(rr, req)
+
+			assert.Equal(t, tc.result, rr.Code)
+			useCase.AssertExpectations(t)
+		})
+	}
+}
+
+// nolint:funlen
+func TestNewGetReportsRevenueHandler_ResponseStructure(t *testing.T) {
+	t.Run("should return correct response structure with all gas tracking fields", func(t *testing.T) {
+		useCase := &MockGetRevenueReportUseCase{}
+		useCase.On("Run", mock.Anything, mock.Anything, mock.Anything).
+			Return(reports.GetRevenueReportResult{
+				TotalQuoteCallFees:    entities.NewWei(1000),
+				TotalGasFeesCollected: entities.NewWei(500),
+				TotalGasSpent:         entities.NewWei(300),
+				TotalPenalizations:    entities.NewWei(100),
+				TotalProfit:           entities.NewWei(1100),
+			}, nil).Once()
+
+		req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "/reports/revenue?startDate=2024-01-01&endDate=2025-08-27", nil)
+		require.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+		handler := handlers.NewGetReportsRevenueHandler(useCase)
+		handler.ServeHTTP(rr, req)
+
+		assert.Equal(t, http.StatusOK, rr.Code)
+
+		var response pkg.GetRevenueReportResponse
```

Comment:
just a tip: you can assert the json directly if you prefer (no need to change anything here)